### PR TITLE
[1.11] Add Mesos patches to ensure TEARDOWN is sent in v1 Java shim.

### PR DESCRIPTION
**Pull request by rukletsov**
## High-level description

Add Mesos patches to ensure TEARDOWN is sent in v1 Java shim.

Destruction of the v1 library used in the v1 Java scheduler shim
is not always under our control. For example, the JVM can call JNI
`finalize()` if the Java scheduler nullifies its reference to the
V1Mesos library immediately after sending `TEARDOWN`. We want to make
sure that `TEARDOWN` is sent before the Mesos library is destructed.

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

File: packages/mesos/extra/patches/0024-Waited-for-TEARDOWN-to-be-sent-in-v1-Java-scheduler-.patch [hunk 1]
```diff
@@ -0,0 +1,52 @@
+From dc0477937b92439f7ac2594589db8cbf08685e78 Mon Sep 17 00:00:00 2001
+From: Alexander Rukletsov <alexr@apache.org>
+Date: Tue, 9 Oct 2018 18:22:16 +0200
+Subject: [PATCH] Waited for TEARDOWN to be sent in v1 Java scheduler shim.
+
+Destruction of the library is not always under our control. For
+example, the JVM can call JNI `finalize()` if the Java scheduler
+nullifies its reference to the V1Mesos library immediately after
+sending `TEARDOWN`. We want to make sure that `TEARDOWN` is sent
+before the Mesos library is destructed.
```
[the remark this answers is not in the document]
Comment (rukletsov) [reply packages/mesos/extra/patches/0024-Waited-for-TEARDOWN-to-be-sent-in-v1-Java-scheduler-.patch]: No it is not a cherry-pick. Those patches are completely different (the one in the master uses `mesos->mesos->call()` with await on the future, which is not available in 1.5.x. Hence here an uglier version of the fix is done, using `sleep()`. This is also the reason why this is not back ported onto 1.5.x.

File: packages/mesos/extra/patches/0024-Waited-for-TEARDOWN-to-be-sent-in-v1-Java-scheduler-.patch [hunk 2]
```diff
@@ -0,0 +1,52 @@
+From dc0477937b92439f7ac2594589db8cbf08685e78 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (rukletsov) [reply packages/mesos/extra/patches/0024-Waited-for-TEARDOWN-to-be-sent-in-v1-Java-scheduler-.patch]: See comment below.

File: packages/mesos/extra/patches/0023-Put-TerminateEvent-at-the-end-of-the-queue-in-the-Me.patch [hunk 3]
```diff
@@ -0,0 +1,38 @@
+From 5e56ee627019b035e736ffa166a8473eeabdc5f0 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (rukletsov) [reply packages/mesos/extra/patches/0023-Put-TerminateEvent-at-the-end-of-the-queue-in-the-Me.patch]: See comment below.

**Comment by rukletsov** [comment]
I feel like this change takes way too long to get reviewed. The original PR has been created more than 2 months ago and was stalling several times. Soliciting reviews, explaining people again and again what is going on here wears me out. What am I going to do now is to yolo this patch in; I feel confident about the fix.

**Comment by rukletsov** [comment]
@mesosphere-mergebot merge-it

**Comment by rukletsov** [comment]
@mesosphere-mergebot merge-it

**Comment by rukletsov** [comment]
@mesosphere-mergebot merge-it

**Comment by rukletsov** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
Merged this directly as all status checks were green.

**Pull request merged**